presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestJoinEnumerator.java

```diff
@@ -112,7 +117,48 @@ public void testDoesNotCreateJoinWhenPartitionedOnCrossJoin()
                 new CostComparator(1, 1, 1),
                 idAllocator,
                 multiJoinNode.getFilter(),
-                noLookup());
+                noLookup(),
+                new Rule.Context()
```

Comment:
This method is very long and 99% of it is boiler plate. Consider extracting helper method to build the context.

```
   Rule.Context context = makeContext();

   ReorderJoins.JoinEnumerator joinEnumerator = new ReorderJoins.JoinEnumerator(
                context.getSession(),
                context.getCostProvider(),
                new CostComparator(1, 1, 1),
                context.getIdAllocator(),
                multiJoinNode.getFilter(),
                context.getLookup(),
                context);



    private Rule.Context makeContext()
    {
        return new Rule.Context()
        {
            private PlanNodeIdAllocator planNodeIdAllocator = new PlanNodeIdAllocator();
            private SymbolAllocator symbolAllocator = new SymbolAllocator();
            private CachingStatsProvider statsProvider = new CachingStatsProvider(
                    queryRunner.getStatsCalculator(),
                    Optional.empty(),
                    noLookup(),
                    queryRunner.getDefaultSession(),
                    symbolAllocator::getTypes);
            private CachingCostProvider costProvider = new CachingCostProvider(
                    queryRunner.getCostCalculator(),
                    statsProvider,
                    Optional.empty(),
                    noLookup(),
                    queryRunner.getDefaultSession(),
                    symbolAllocator::getTypes);

            @Override
            public Lookup getLookup()
            {
                return noLookup();
            }

            @Override
            public PlanNodeIdAllocator getIdAllocator()
            {
                return planNodeIdAllocator;
            }

            @Override
            public SymbolAllocator getSymbolAllocator()
            {
                return symbolAllocator;
            }

            @Override
            public Session getSession()
            {
                return queryRunner.getDefaultSession();
            }

            @Override
            public StatsProvider getStatsProvider()
            {
                return statsProvider;
            }

            @Override
            public CostProvider getCostProvider()
            {
                return costProvider;
            }

            @Override
            public void checkTimeoutNotExhausted() {}
        };
    }
```